modules/EvseV2G/connection.cpp

```diff
@@ -985,3 +1022,44 @@ int connection_start_servers(struct v2g_context* ctx) {
 
     return 0;
 }
+int create_udp_socket(const u_int16_t udp_port) {
+    // find suitable addresses
+    addrinfo hints;
+    addrinfo* result;
+
+    memset(&hints, 0, sizeof(addrinfo));
+    hints.ai_family = AF_INET6;     // allow only IPv6
+    hints.ai_socktype = SOCK_DGRAM; // Datagram socket
+    hints.ai_flags = AI_ALL;
+    hints.ai_protocol = 0; // OS selects best protocol (best practise)
+
+    const auto port_str = std::to_string(udp_port);
+
+    //                                    alternatively:
+    //                                    "localhost"
+    const auto get_addrinfo_result = getaddrinfo(nullptr, port_str.c_str(), &hints, &result);
+
+    if (get_addrinfo_result != 0) {
+        dlog(DLOG_LEVEL_ERROR, "getaddrinfo: %s", gai_strerror(get_addrinfo_result));
+        return -1;
+    }
+
+    if (result == nullptr) {
+        dlog(DLOG_LEVEL_ERROR, "Could not get address info");
+        return -1;
+    }
+
+    const auto udp_socket = socket(result->ai_family, result->ai_socktype, result->ai_protocol);
+    if (udp_socket == -1) {
+        dlog(DLOG_LEVEL_ERROR, "Could not create socket");
+        return -1;
+    }
+    const auto connected = connect(udp_socket, result->ai_addr, result->ai_addrlen);
```

Comment:
The TLS key extraction of the dSPACE Wireshark plugin requires a particular source port:

> The disclosure message is a UDP packet within the source port range 49152-65535

See also https://github.com/dspace-group/dsV2Gshark/blob/6da7c8f673c367a91a15d2da2ae9a9ea8dc68a9a/Wireshark/plugins/v2gtlssecret.lua#L207.

You need to `bind()` your socket to such a source port for this particular plugin to work. (The payload should work, at first glance.)

Untested, just from reading the plugin.

Reply:
That is a good point. I was experimenting with an udp socket client yesterday and unfortunately the OS sometimes selects ports that are not in the dynamic port radius (49152-65535). So we should bind a random port (between 49152 and 65535) for the UDP client as well.